src/Bootstrap/Carousel.elm

```diff
@@ -0,0 +1,509 @@
+
+                    Start transition ->
+                        [ ( "active", True )
+                        ]
+
+                    Animating transition ->
+                        let
+                            { directionalClassName } =
+                                classNames transition
+                        in
+                            [ ( "active", True )
+                            , ( directionalClassName, True )
+                            ]
+            else if index == newIndex then
+                case tstage of
+                    NotAnimating ->
+                        []
+
+                    Start transition ->
+                        [ ( .orderClassName (classNames transition), True )
+                        ]
+
+                    Animating transition ->
+                        let
+                            { directionalClassName, orderClassName } =
+                                classNames transition
+                        in
+                            [ ( directionalClassName, True )
+                            , ( orderClassName, True )
+                            ]
+            else
+                []
+    in
+        slide
+            |> Slide.removeActive
+            |> Slide.addAttributes [ classList classes ]
+            |> Slide.view
+
+
+slides : List (Slide.Config msg) -> Config msg -> Config msg
+slides newSlides (Config settings) =
+    Config { settings | slides = newSlides }
+
+
+withIndicators : Config msg -> Config msg
+withIndicators (Config settings) =
+    Config { settings | indicators = True }
+
+
+withControls : Config msg -> Config msg
+withControls (Config settings) =
+    Config { settings | controls = True }
+
+
+controlPrev : String -> Html.Html Msg
+controlPrev id =
+    a [ class "carousel-control-prev", attribute "role" "button", onClick (StartTransition Prev) ]
+        [ span [ class "carousel-control-prev-icon", attribute "aria-hidden" "true" ] []
+        , span [ class "sr-only" ] [ text "Previous" ]
+        ]
+
+
+controlNext : String -> Html.Html Msg
+controlNext id =
+    a [ class "carousel-control-next", attribute "role" "button", onClick (StartTransition Next) ]
+        [ span [ class "carousel-control-next-icon", attribute "aria-hidden" "true" ] []
+        , span [ class "sr-only" ] [ text "Next" ]
+        ]
+
+
+indicators : String -> Int -> Int -> Html.Html Msg
+indicators id size activeIndex =
+    let
+        item n =
+            Html.li
+                [ classList [ ( "active", n == activeIndex ) ]
+                , onClick (StartTransition (Number n))
+                ]
+                []
+
+        items =
+            List.range 0 (size - 1)
+                |> List.map item
+    in
+        Html.ol [ class "carousel-indicators" ] items
+
+
+type CarouselOption msg
+    = Attrs (List (Html.Attribute msg))
+
+
+config : String -> (Msg -> msg) -> List (CarouselOption msg) -> Config msg
```

Comment:
The options argument can be removed and you can set attributes as part of a config "pipeline". (see attrs comment)